pkg/agent/netpol/netpol.go

```diff
@@ -55,6 +59,12 @@ func Run(ctx context.Context, nodeConfig *config.Node) error {
 		return err
 	}
 
+	// As kube-router netpol requires addresses to be available in the node object
+	// Loop until the node has ready addresses to avoid race conditions (max 1 minute).
+	ctx, cancel := context.WithTimeout(context.Background(), 60*time.Second)
+	defer cancel()
+	checkNodeTaint(ctx, client, nodeConfig.AgentConfig.NodeName)
```

Comment:
You should be able to replace both this context and the checkNodeTaints with [wait.PollImmediateUntilWithContext](https://pkg.go.dev/k8s.io/apimachinery@v0.26.0/pkg/util/wait#PollImmediateUntilWithContext), or [wait.PollImmediateWithContext](https://pkg.go.dev/k8s.io/apimachinery@v0.26.0/pkg/util/wait#PollImmediateWithContext) if you want to put a limit on how long it waits.

A good example of using this to wait for the local node's status is at https://github.com/k3s-io/k3s/blob/941119640612808cfb2d71d1c85bec74740b493a/pkg/agent/tunnel/tunnel.go#L165-L190

Reply:
They are deprecated but is still sees wide use across the Kubernetes codebase and within our projects. The difference between the deprecated and newer methods is mostly around how they return context errors. I'd prefer to stick with the current functions until we are ready to move the entire codebase off `PollImmediateUntilWithContext`/`PollImmediateWithContext`.

Reply:
Ok. But let me add a comment about it